File: 프로그래머스/unrated/181916. 주사위 게임 3/주사위 게임 3.js
function solution(a, b, c, d) {
    let answer = 0;
    const count = new Set([a, b, c, d]);
    const array = new Array(7);
    
    array.fill(0);
    
    array[a]++;
    array[b]++;
    array[c]++;
    array[d]++;
    
    const three = array.indexOf(3);
    const twof = array.indexOf(2);
    const twol = array.lastIndexOf(2);
    const onef = array.indexOf(1);
    const onel = array.lastIndexOf(1);
    
    switch(count.size) {
        case 1 : answer = 1111 * a; break;
        case 2 :
            if (three !== -1) answer = (10 * three + onef) ** 2;
            else answer = (twof + twol) * Math.abs(twof - twol);
            break;
        case 3 : answer = onef * onel; break;
        case 4 : answer = Math.min(a, b, c, d); break;
    }
    return answer;
}

File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.java
import java.util.Arrays;
class Solution {
    public int solution(int a, int b, int c, int d) {
        int[] arr = {a,b,c,d};
        int answer = Arrays.stream(arr).min().getAsInt();
        int p = 0;
        int q = 0;
        int r = 0;
        for(int i = 0; i < arr.length; i++){
            int count = 0;
            q = 0;
            for(int j = 0; j < arr.length; j++){
                if(i==j){continue;}
                if(arr[i] == arr[j]){
                    p = arr[j];
                    count++;
                }
                if(q == 0 && arr[i] != arr[j]){
                    q = arr[j];
                }
                if(arr[i] != arr[j] && q != arr[j]){
                    r = arr[j];
                }
            }
            switch (count) { 

            case 3: answer = p * 1111;
                    break;
            case 2: answer = (10 * p + q)*(10 * p + q);
                    break;
            case 1: answer = (p + q) * Math.abs(p - q);
                    if(r!=0){
                        answer = q * r;
                    }
                    break;
            }
        }
        return answer;
    }
}